connect/mirror/src/main/java/org/apache/kafka/connect/mirror/MirrorCheckpointConnector.java

```diff
@@ -262,12 +264,13 @@ private void createInternalTopics() {
         );
     }
 
-    Map<TopicPartition, OffsetAndMetadata> listConsumerGroupOffsets(String group)
+    ListConsumerGroupOffsetsResult listConsumerGroupOffsets(List<String> groups)
             throws InterruptedException, ExecutionException {
+        Map<String, ListConsumerGroupOffsetsSpec> groupSpecs = groups.stream()
+                .collect(Collectors.toMap(group -> group, group -> new ListConsumerGroupOffsetsSpec()));
         return adminCall(
-                () -> sourceAdminClient.listConsumerGroupOffsets(group).partitionsToOffsetAndMetadata().get(),
-                () -> String.format("list offsets for consumer group %s on %s cluster", group,
-                        config.sourceClusterAlias())
+                () -> sourceAdminClient.listConsumerGroupOffsets(groupSpecs),
+                () -> String.format("list offsets for consumer groups %s on %s cluster", groups, config.sourceClusterAlias())
         );
```

Comment:
I think to make this adminCall functional, the first argument needs to be blocking. I think you need to use .all().get() instead of returning the whole Result object.

Reply:
While looping, we need to get topic partitions and metadata by group. For this reason, I have chosen `ListConsumerGroupOffsetsResult` as the return of listConsumerGroupOffsets as it allows us to do `result.partitionsToOffsetAndMetadata(group).get()`.

In `.all().get()` approach, once we have all topic partitions and metadata for all consumer groups, we will not be able to distinguish each of them by group.

Let me know if I have missed something.

Many thanks for your review, @gharris1727! :)

Reply:
`ListConsumerGroupOffsetsResult#all()` has the return type `KafkaFuture<Map<String, Map<TopicPartition, OffsetAndMetadata>>>` which should include group information. I don't think it merges the results of the different groups together.

Reply:
Oh! You are right! Sorry, Greg, I have mixed up the return type. I will push a commit to fix it. Constructive review, thanks!